lib/Date.php

```diff
@@ -47,10 +47,12 @@ public static function fromYearMonthDayString(string $value): self
             throw new InvalidDate('Failed to parse string as a Y-m-d formatted date.');
         }
 
+        [, $year, $month, $day] = $matches;
```

Comment:
Infection changes `$matches[1]` to `$matches[0]`, which is the entire match string, and `2020-01-02` casts to `2020` as an int so is uncaught but also impossible to detect, unless I'm missing something. Seems overkill to be validating each value in the matches is an integer due to the regex pattern already doing this.